packages/xstate-wallet/src/workflows/virtual-funding-as-hub.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  StateNodeConfig,
+  MachineConfig,
+  Machine,
+  MachineOptions,
+  AnyEventObject,
+  DoneInvokeEvent,
+  ServiceConfig,
+  assign,
+  spawn
+} from 'xstate';
+import {filter, flatMap} from 'rxjs/operators';
+
+import {Store} from '../store/memory-store';
+import {LedgerFunding, VirtualFundingAsLeaf} from '.';
+import {checkThat} from '../utils';
+import {isSimpleEthAllocation} from '../utils/outcome';
+
+import {FundGuarantor, AllocationItem, isFundGuarantor, Participant} from '../store/types';
+
+import _ from 'lodash';
+import {Role, States} from './virtual-funding-as-leaf';
+
+type Init = VirtualFundingAsLeaf.Init;
+
+type Deductions = {
+  deductions: {
+    [Role.A]: AllocationItem[];
+    [Role.B]: AllocationItem[];
+  };
+};
+
+type WithDeductions = Init & Deductions;
+
+type TEvent = AnyEventObject;
+
+const enum Actions {
+  assignDeductions = 'assignDeductions',
+  watchObjectives = 'watchObjectives'
+}
+
+const enum Services {
+  getDeductions = 'getDeductions',
```

Comment:
Not sure I fully understand what deductions are. What is the benefit over current allocation items + desired allocation items?